app/models/logstash_queries/top_n_modules_query.rb

```diff
@@ -1,22 +1,22 @@
-class TopNModulesQuery < TopNQuery
+# frozen_string_literal: true
 
+class TopNModulesQuery < TopNQuery
   def booleans(json)
-    json.must do
-      json.child! { json.term { json.affiliate @affiliate_name } }
+    json.filter do
+      json.child! { json.term { json.set! 'params.affiliate', @affiliate_name } }
       modules_must(json)
       additional_musts(json)
     end
     json.must_not do
-      json.child! { json.term { json.set! "useragent.device", "Spider" } }
-      json.child! { json.term { json.raw "" } }
+      json.child! { json.term { json.set! 'useragent.device', 'Spider' } }
+      json.child! { json.term { json.set! 'params.query.raw', '' } }
```

Comment:
@MothOnMars It is not recommended  to use .raw on params. It can open us up to security vulnerabilities.  sanitize is a safer way to go.

Reply:
The `params` in this line are not ActionController params. They refer to specific fields in an Elasticsearch document:
```
"params": {
            "query": "mars curiosity rover",
```

Reply:
ok in the future best to rename that something different than params. Maybe query_params